kv/dist_sender.go

```diff
@@ -707,6 +721,13 @@ func (ds *DistSender) sendChunk(ctx context.Context, ba roachpb.BatchRequest) (*
 		// Immediately return if querying a range failed non-retryably.
 		if pErr != nil {
 			return nil, pErr, false
+		} else if !finished {
+			select {
+			case <-ds.rpcRetryOptions.Closer:
+				return nil, roachpb.NewErrorf("shutting down"), false
```

Comment:
we have a `NodeUnavailableError` somewhere. Not clear what the semantics should be though.